tests/unit/downloader/test_xls_generation.py

```diff
@@ -0,0 +1,140 @@
+                    'project.contributors.institution': 'University of Prague',
+                    'project.contributors.laboratory': 'Department of Neurology',
+                    'project.contributors.address': '123 Test Street',
+                    'project.contributors.country': 'UK',
+                    'project.contributors.corresponding_contributor': 'yes',
+                    'project.contributors.project_role.text': 'experimental scientist',
+                    'project.contributors.project_role.ontology': 'EFO:0009741',
+                    'project.contributors.project_role.ontology_label': 'experimental scientist',
+                    'project.contributors.orcid_id': 'https://orcid.org/1234-5678-9012-3456'
+                },
+                {
+                    'project.contributors.name': 'Lady Carneval',
+                    'project.contributors.email': 'ladyc@gott.com',
+                    'project.contributors.institution': 'University of Prague',
+                    'project.contributors.laboratory': 'Department of Neurology',
+                    'project.contributors.address': '123 Test Street',
+                    'project.contributors.country': 'UK',
+                    'project.contributors.corresponding_contributor': 'yes',
+                    'project.contributors.project_role.text': 'experimental scientist',
+                    'project.contributors.project_role.ontology': 'EFO:0009741',
+                    'project.contributors.project_role.ontology_label': 'experimental scientist',
+                    'project.contributors.orcid_id': 'https://orcid.org/9999-9999-9999-9999'
+                },
+                {
+                    'project.contributors.name': 'Baby Shark',
+                    'project.contributors.email': 'sharkb@gott.com',
+                    'project.contributors.institution': 'University of Cambridge',
+                    'project.contributors.laboratory': 'Department of Neurology',
+                    'project.contributors.address': '123 Ocean Drive',
+                    'project.contributors.country': 'USA',
+                    'project.contributors.corresponding_contributor': 'no',
+                    'project.contributors.project_role.text': 'data wrangler',
+                    'project.contributors.project_role.ontology': 'EFO:0009737',
+                    'project.contributors.project_role.ontology_label': 'data curator',
+                    'project.contributors.orcid_id': 'https://orcid.org/0000-0000-0000-0001'
+                },
+            ]
+        }
+
+        # when
+        workbook: Workbook = self.downloader.create_workbook(input_json)
+
+        #expect
+        self.assertEqual(len(workbook.worksheets), 1)
+
+        self.__assert_sheet(workbook, contributors_sheet_title, input_json)
+
+    def test_given_input_has_many_rows_and_many_sheets_of_data_successfully_creates_a_workbook(self):
+        # given
+        project_sheet_title = 'Project'
```

Comment:
You can make this a list, and change the assertions to use the list.

Reply:
@amnonkhen I can't see how a list would simplify this. Even a `dict` would make it more complex.
Could you advise what you meant with your comment, please?

Reply:
Sure.

```python
sheet_titles = ['Project', 'Project - Contributors', 'Project - Publications', 'Project - Funders']
# now add the 'when' bits
map(lambda title: self.__assert_sheet(workbook, title, input_json), sheet_titles)
```

Reply:
Tx. I am going to apply it.